src/function.rs

```diff
@@ -205,6 +249,21 @@ pub struct PropertyCallbackInfo {
   args: *mut Opaque,
 }
 
+#[repr(C)]
+#[derive(Debug)]
+pub struct IntegerPropertyCallbackInfo(PropertyCallbackInfo);
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct BooleanPropertyCallbackInfo(PropertyCallbackInfo);
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct ArrayPropertyCallbackInfo(PropertyCallbackInfo);
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct PropertyAttributePropertyCallbackInfo(PropertyCallbackInfo);
```

Comment:
Generally we try to stick as close to the V8 API as possible. Understand that you're providing better types here - but I think I'd rather keep it as minimal as possible in terms of introducing new symbols.

Reply:
Thanks for the feedback @ry, for me,  was a bit weird using `AccessorNameGetterCallback` for the `query` or `deleted` interceptors, but I get why you want to have it close to the V8 API, I've just pushed the changes to remove all these custom types